spec/rubocop/config_loader_spec.rb

```diff
@@ -440,6 +440,99 @@
       end
     end
 
+    context 'when a department is disabled' do
+      let(:file_path) { '.rubocop.yml' }
+
+      shared_examples 'resolves enabled/disabled for all cops' do
+        |enabled_by_default, disabled_by_default|
+        it "handles EnabledByDefault: #{enabled_by_default}, " \
+           "DisabledByDefault: #{disabled_by_default}" do
+          create_file('grandparent_rubocop.yml', <<~YAML)
+            Metrics/AbcSize:
+              Enabled: true
+
+            Metrics/PerceivedComplexity:
+              Enabled: true
+
+            Lint:
+              Enabled: false
+          YAML
+          create_file('parent_rubocop.yml', <<~YAML)
+            inherit_from: grandparent_rubocop.yml
+
+            Metrics:
+              Enabled: false
+
+            Metrics/AbcSize:
+              Enabled: false
+          YAML
+          create_file(file_path, <<~YAML)
+            inherit_from: parent_rubocop.yml
+
+            AllCops:
+              EnabledByDefault: #{enabled_by_default}
+              DisabledByDefault: #{disabled_by_default}
+
+            Style:
+              Enabled: false
+
+            Metrics/MethodLength:
+              Enabled: true
+
+            Metrics/ClassLength:
+              Enabled: false
+
+            Lint/RaiseException:
+              Enabled: true
+
+            Style/AndOr:
+              Enabled: true
+          YAML
+
+          def enabled?(cop)
+            configuration_from_file.for_cop(cop)['Enabled']
+          end
+
+          # Department disabled in parent config, cop enabled in child.
+          expect(enabled?('Metrics/MethodLength')).to be(true)
+
+          # Department disabled in parent config, cop disabled in child.
+          expect(enabled?('Metrics/ClassLength')).to be(false)
+
+          # Enabled in grandparent config, disabled in parent.
+          expect(enabled?('Metrics/AbcSize')).to be(false)
+
+          # Enabled in grandparent config, department disabled in parent.
+          expect(enabled?('Metrics/PerceivedComplexity')).to be(false)
+
+          # Pending in default config, department disabled in grandparent.
+          expect(enabled?('Lint/StructNewOverride')).to be(false)
+
+          # Department disabled in child config.
+          expect(enabled?('Style/Alias')).to be(false)
+
+          # Department disabled in child config, cop enabled in child.
+          expect(enabled?('Style/AndOr')).to be(true)
+
+          # Pending in default config, department disabled in grandparent,
+          # cop enabled in child config.
+          expect(enabled?('Lint/RaiseException')).to be(true)
+
+          # Pending in default config.
+          expect(enabled?('Layout/SpaceAroundMethodCallOperator'))
+            .to eq(enabled_by_default ||
+                   (disabled_by_default ? false : 'pending'))
```

Comment:
The pending cops state depends on production config/default.yml. This test seems to fail when pending cops will be enabled by default in RuboCop 1.0.

So, the test will fail if config/default.yml would be edited as follows:

```diff
diff --git a/config/default.yml b/config/default.yml
index 9c25d5ee8..38ec16d61 100644
--- a/config/default.yml
+++ b/config/default.yml
@@ -1107,7 +1107,7 @@ Layout/SpaceAroundKeyword:

 Layout/SpaceAroundMethodCallOperator:
   Description: 'Checks method call operators to not have spaces around them.'
-  Enabled: pending
+  Enabled: true
   VersionAdded: '0.82'
```

Reply:
Yes, I think we can just update the test when we change the default for that cop.

Reply:
I think there are two problems.

- This test case check would not be possible when there is no pending cops (For example, immediately after a major release)
- This test may break with a major release

Unfortunately I don't yet have a way to solve these 💦 Anyway, this PR feature is very user-friendly and a great implementation 👍

Reply:
This small part of the spec example is not very important. I'll remove it to save us future headache. :slightly_smiling_face: